### ISSUE TITLE
turf.difference returns null when polygon has hole

### ISSUE THREAD

**Issue by camhart** [Turfjs/turf#1834]
If the polygon you substract from has a hole in it, the result is null.

Run this in nodejs, using turf `5.1.6`.

```
const turf = require('@turf/turf')

let areaToWhiteout = turf.polygon(
	[
      [
        [
          0, -8
        ],
        [
          1920,
          -8
        ],
        [
          1920,
          1040
        ],
        [
          0,
          1040
        ],
        [
          0,
          -8
        ]
      ]
    ]
)
	
let areaToBlackoutReturnsNull = turf.polygon(
	[
      [
        [
          1920,
          -8
        ],
        [
          1920,
          1040
        ],
        [
          3840,
          1040
        ],
        [
          3840,
          -8
        ],
        [
          1920,
          -8
        ]
      ],
      [
        [
          2322,
          148
        ],
        [
          3469,
          148
        ],
        [
          3469,
          965
        ],
        [
          2322,
          965
        ],
        [
          2322,
          148
        ]
      ]
	]
)

let areaToBlackoutWorks = turf.polygon(
	[
      [
        [
          1920,
          -8
        ],
        [
          1920,
          1040
        ],
        [
          3840,
          1040
        ],
        [
          3840,
          -8
        ],
        [
          1920,
          -8
        ]
      ]
	]
)

let diffWithIsland = turf.difference(areaToBlackoutReturnsNull, areaToWhiteout)
let diffWithoutIsland = turf.difference(areaToBlackoutWorks, areaToWhiteout)

console.log(diffWithIsland) //returns null when it shouldn't
console.log(diffWithoutIsland) //returns difference (accurately)
```

As https://github.com/Turfjs/turf/issues/1661 recommends, I've tried cleanCoords and unkink--neither did anything.  I am not using turf with geo data though--just a coordinate grid (x/y) to figure out areas of a screenshot I want to blackout based off window coordinates.  I *hope* this doesn't matter.

**Comment by camhart** [comment]
This appears to be related to https://github.com/Turfjs/turf/issues/1835

**Comment by rowanwins** [comment]
Resolved in v6.3.0
